crates/bevy_ui/src/render/mod.rs

```diff
@@ -645,43 +690,17 @@ pub struct UiBatch {
 const TEXTURED_QUAD: u32 = 0;
 const UNTEXTURED_QUAD: u32 = 1;
 
-#[allow(clippy::too_many_arguments)]
-pub fn queue_uinodes(
-    extracted_uinodes: Res<ExtractedUiNodes>,
-    ui_pipeline: Res<UiPipeline>,
-    mut pipelines: ResMut<SpecializedRenderPipelines<UiPipeline>>,
-    mut views: Query<(&ExtractedView, &mut RenderPhase<TransparentUi>)>,
-    pipeline_cache: Res<PipelineCache>,
-    draw_functions: Res<DrawFunctions<TransparentUi>>,
-) {
-    let draw_function = draw_functions.read().id::<DrawUi>();
-    for (view, mut transparent_phase) in &mut views {
-        let pipeline = pipelines.specialize(
-            &pipeline_cache,
-            &ui_pipeline,
-            UiPipelineKey { hdr: view.hdr },
-        );
-        transparent_phase
-            .items
-            .reserve(extracted_uinodes.uinodes.len());
-        for (entity, extracted_uinode) in extracted_uinodes.uinodes.iter() {
-            transparent_phase.add(TransparentUi {
-                draw_function,
-                pipeline,
-                entity: *entity,
-                sort_key: FloatOrd(extracted_uinode.stack_index as f32),
-                // batch_size will be calculated in prepare_uinodes
-                batch_size: 0,
-            });
-        }
-    }
-}
-
 #[derive(Resource, Default)]
 pub struct UiImageBindGroups {
     pub values: HashMap<Handle<Image>, BindGroup>,
 }
 
+pub fn queue_uinodes(mut extracted_uinodes: ResMut<ExtractedUiNodes>) {
+    extracted_uinodes
+        .ranges
+        .sort_unstable_by_key(|extracted_range| extracted_range.sort_key);
```

Comment:
Did you try radsort?

Reply:
I'll try it out. 

I made an alternative implementation as well that doesn't require a sort. Instead of storing the layout geometry in components on the uinode entities it has a single layout entity that has a list of all the uinode rects in a vector. It might be a better approach and the prototype seems quite promising but I don't have most of the systems like clipping, interaction, etc working yet. Everything needs to be rewritten to support it.

Reply:
With rad sort:
<img width="491" alt="queue-rad-sort" src="https://github.com/bevyengine/bevy/assets/27962798/0cfa611f-2f32-4c1d-955d-6d1a183d6dae">

Reply:
Which is which? What is being compared?